New RSTUF Guide (#341)
repository-service-tuf/repository-service-tuf: docs/source/guide/general/usage.rst

asynchronous process.

Submitting a request for adding/removing will return an HTTP status code 202,
meaning the task is submitted. As part of the body, there is the ``task id``.

Response body adding/removing targets, ``task id``:

.. code-block:: json

    {
        "data": {
            "task_id": "<TASK ID>",
        },
    }

This body has more details such as targets, last_update, etc. See the
`endpoint response documentation <https://repository-service-tuf.github.io/repository-service-tuf-api/>`_

To retrieve the task status, send requests to
``http://<IP ADDRESS>/api/v1/task?task_id=<TASK ID>`` with method
``GET``.

The response will have a body with all task details

.. code-block:: json

    {
        "data": {
            "task_id": "<TASK ID>",
            "state": "<STATE>",
            "result": {
            "status": "Task finished.",
            "details": {
                "key": "value"
            }
            }
        },
        "message": "Task state."
    }

See the `endpoint task documentation <https://repository-service-tuf.github.io/repository-service-tuf-api/#/v1/get_api_v1_task__get>`_
for more details.

> Adding targets/artifacts for back-singing

Reviewer: Do you mean "back-signing"?

Normally, when you add a target through the API there will be an overhead of
multiple additional operations which for a large number of targets can prove
to be significant.

Here are some benchmarks of the "import targets" feature:

* Running in a Macbook Pro (2019) 2,4 GHz 8-Core Intel Core i9/32GB 2667 MHz DDR4:

    - Adding 500,000 targets: ~40 minutes
       + Loaded 1 of 1 file with 500,000 targets
    - Adding 1,000,000 targets: ~55 minutes
       + Loaded 1 of 2 file with 500,000 targets
       + Loaded 2 of 2 file with 500,000 targets

.. warning::

    Use the API flow integration to your release process (CI/CD or Distribution
    Platform).

    Do not use "import targets" as a replacement for the standard procedure to
    add targets throughout the RSTUF API or CLI tool after RSTUF is deployed.
    This feature should only be used before going live with RSTUF.

RSTUF-CLI contains the :ref:`guide/repository-service-tuf-cli/index:Import Targets  (``import-targets\`\`)` feature.

CLI usage
---------

.. include:: ../repository-service-tuf-cli/index.rst
    :start-after: rstuf-cli-admin-import-targets
    :end-before: rstuf-cli-key

.. note::

>     If you provide users with download or update tools, you need to add
>     functionality to your tools to check the signed metadata.

Reviewer: ```suggestion
    When providing users with download or update capabilities, it is necessary to add
    signed metadata checkpoints to the functionality tools.
```
